extensions/wikia/Helios/HelperController.class.php

```diff
@@ -127,4 +130,49 @@ public function sendConfirmationEmail()
 
 		$this->response->setVal( 'success', true );
 	}
+
+	public function isBlocked() {
+		if ( !$this->authenticateViaTheSchwartz() ) {
+			return;
+		}
+
+		$username = $this->getFieldFromRequest( 'username', 'invalid username' );
+		if ( !isset( $username ) ) {
+			return;
+		}
+
+		$blocked = $this->getAuthService()->isUsernameBlocked( $username );
+
+		$this->response->setData( array( 'blocked' => $blocked ) );
+	}
+
+	protected function getFieldFromRequest( $field, $failureMessage ) {
+		$username = $this->getVal( 'username', null );
+		if ( !isset( $username ) ) {
+			$this->response->setVal( 'message', $failureMessage );
+			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+		}
+
+		return $username;
+	}
+
+	protected function authenticateViaTheSchwartz() {
```

Comment:
Would be nice to use this function in other functions in this class where we check The Schwartz Secret Token